src/AAL/AAL_knapsack.cpp

```diff
@@ -1,6 +1,8 @@
 #include "AAL_knapsack.h"
 
 #include <algorithm>
+#include <cmath>
+#include <exception>
```

Comment:
```suggestion
#include <stdexcept>
```
`std::runtime_error` is part of `<stdexcept>`. The standard `<exception>` header does not define `std::runtime_error`.